spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,280 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+introspection system.
+
+For example, a type `StringFilter` could be described as:
+
+```graphql example
+tagged StringFilter {
+  contains: String!
+  lengthAtLeast: Int!
+  lengthAtMost: Int!
+}
+```
+
+In this case we're representing exactly one of the following:
+
+- an object containing a single key `contains` with a {String} value
+- an object containing a single key `lengthAtLeast` with an {Int} value
+- an object containing a single key `lengthAtMost` with an {Int} value
+
+The `StringFilter` Tagged type is valid as both an input and output type, but
+this is not true of all tagged types. If a Tagged type has a member whose type
+is only valid for output, then the Tagged type is only valid for output. If a
+Tagged type has a member whose type is only valid for input, then the Tagged
+type is only valid for input. If a Tagged type has a member that is only valid
+for input, and another member that's only valid for output, then it is not a
+valid Tagged type.
+
+A member of a Tagged type may be a Scalar, Enum, Input Object type, Object
+type, an Interface, a Union or another Tagged type. Additionally, it may be
+any wrapping type (e.g. list, non-null, or any combination thereof) whose
+underlying base type is one of those six.
+
+Selecting all the members of our `StringFilter` type:
+
+```graphql example
+{
+  contains
+  lengthAtLeast
+  lengthAtMost
+}
+```
+
+Could yield one of the following objects:
+
+- `{ "contains": "Awesome" }`
+- `{ "lengthAtLeast": 3 }`
+- `{ "lengthAtMost": 42 }`
+
+Valid queries must supply a nested field set for a field that returns a Tagged
+type, so for this schema:
+
+```graphql example
+type Query {
+  stringFilter: StringFilter
+}
+```
+
+This query is not valid:
+
+```graphql counter-example
+{
+  stringFilter
+}
+```
+
+However, this query is valid:
+
+```graphql example
+{
+  stringFilter {
+    contains
+  }
+}
+```
+
+And may yield one of the following objects:
+
+- `{ "contains": "Awesome" }`
+- `{}`
+
+**Field Ordering**
+
+Like with Object types, when querying a Tagged type, the resulting mapping of
+fields are conceptually ordered in the same order in which they were
+encountered during query execution, excluding fragments for which the type does
+not apply and fields or fragments that are skipped via `@skip` or `@include`
+directives, and fields which are neither the single matched member nor the
+`__typename` introspection field. This ordering is correctly produced when
+using the {CollectFields()} algorithm.
+
+**Result Coercion**
+
+Determining the result of coercing an object or tagged type is the heart of the
+GraphQL executor, so this is covered in that section of the spec. Note that
+only Tagged types for which {IsOutputType(type)} returns {true} may be used in
+output.
+
+**Input Coercion**
+
+Only Tagged types for which {IsInputType(type)} returns {true} may be used as
+input.
+
+The value for an input Tagged type should be an input object literal or an
```

Comment:
```suggestion
The value for an input Tagged type must be an input object literal or an
```

Reply:
This "should" is the same word used in Input Coercion for Input Objects; I think the `should/otherwise ... must` is unambiguous, so I'm going to leave this as is, but I appreciate you flagging it :+1: